proto/one/enterprise/data/enterprise_dataset_definition.proto

```diff
@@ -24,6 +24,9 @@ message EnterpriseDatasetDefinition {
 
     // One or more DataTables that composes the Dataset
     repeated EnterpriseDataTableDefinition dataTables = 6;
+    
+    // DataSource query parameters
+    repeated EnterpriseDataTableQueryParameter queryParameters = 7;
```

Comment:
This is the change from @AlanLiu-AI 's branch right?  If we have the plural protobuf would it be better to use that here or is a list easier to work with?  From experience sometimes the difference between a single plural protobuf and an Enumerable of the singular can be confusing and annoying.  Not sure what the answer is but just raising the question.  I think typically when it is a property of another model we use repeated vs the plural model.

Reply:
That's a great question. I'll have to relay it to @stevenhaching. Any thoughts here?

Reply:
@AQI-DanG :  Is you suggestion to reference the plural here instead of redefining the repeated here?
Doesn't this create a double child?

Reply:
I was just commenting that we could reference the plural, I'm not sure it's a good idea though as then you have to navigate to the items property before you can get to the enumerable.
I think this comment was a result of me being annoyed because I had to convert an enumerable of variable on a computation to a variables object and it is not as  straightforward as I wanted it to be.

Reply:
@AQI-DanG and @stevenhaching what's the best course of action on this?

Reply:
Looking at other precedents it seems like when one model has many of another we use repeated.  The plural definitions are mainly used in the ApiResponse object so I think it was fine the way you had it.  I apologize for the distraction.